### ISSUE TITLE
rider 420 compatible?

### ISSUE THREAD

**Issue by ghost** [andbue/ors2bryton#1]
thanks a lot for the script.

to get it running though, the missing dependency _lxml-python_ had to be installed in first place.

so after planning and exporting a route from openrouteservice as Ors API GPX (instructions included), 3 files were created during the gpxfile.gpx conversion process:

gpxfile.smy
gpxfile.tinfo
gpxfile.track

i gave it a try on my rider 420, unfortunately the option "follow route" fails to fetch the imported route? i test-copied to various folders on the device like "tracks" or "plantrip" without luck.

what am i missing?

**Comment by andbue** [comment]
Hi, unfortunately I've only got the 330 model. I simply copy the smy, tinfo and track files into the Tracks folder, switch on the device, and find the new entry under Follow track > View. Maybe Bryton has removed this functionality in newer models in order to force users to install their annoying and dysfunctional app?

**Comment by ghost** [comment]
yes, that's exactly my approach, but no matter what i try, it keeps telling me "empty".

unfortunately, i am not even aware of the exact format of the _rider 420_, as i never used their strictent awkwarded software to create a tour. a shame to market something as closed and limited as this as user friendly as well.

for days i have been trying to find more details on it (_format_, _destination folder_), no chance. their support is practically non-existant.

if anyone can shed some infos here, this would help a lot.

**Comment by ghost** [comment]
thanks to the help of a kind rider 420 user, i can finally share more details with you.

it seems that planned app routes are currently stored as .fit files in the "plantrip" folder, converted internally on the device itself and then stored in the "tracks" folder:

route-a.smy
route-a.tinfo

additionally a subfolder with the name of the tour (route-a) and the following contents will be created as well:

several .zmap files, depending on the length of the tour
a .tbt.zinfo file, with the tbt informations

shouldn't it theoretically be possible to convert the ors api gpx including instructions to the .fit-format? these could then easily be copied to the "plantrip" folder so to avoid the actual conversion, which should be much more difficult to realize.

unfortunately i can't seem to find a converter that can do this - offline and for linux even less.

**Comment by andbue** [comment]
Good to know that there still is a way to load files without the app! For the .fit format there is a SDK available at https://fit-route.pyrites.org.uk/. I would be surprised if bryton really followed the standard and did not include some custom extensions in their internal conversion. What I would do in your place: try to get a fit file with routing instructions to show up on the device. Maybe you can create these files with the official app and then reconstruct how they have to be written.

When I built the code for the 330, I wrote a file containing waypoints and different instructions, then walked down the street taking notes about what showed up on the display when reaching a waypoint...

**Comment by ghost** [comment]
thanks for referencing the converter.

i placed the converted .fit-file in the plantrip folder, but the rider 420 refuses to make use of it: route incomplete - just as you expected.

besides - assuming the conversion went well and no information got lost, there is a huge difference in filesize. i can hardly tell of the benefits of this format, but is fit-files really that compact?

ors-tbt-gpx: 31,8kib
fit: 2,3kib

regarding app generated routes: i still have the provided files given to me. i guess it would be ok to post them here, if it is of any interest for more advanced users?

**Comment by andbue** [comment]
GPX is a xml file. .fit, .tinfo, .track are binary formats. It's hard to guess from the file size alone if any relevant information is lost by converting.
Feel free to attach any files here! I don't know if there are any more active projects around bryton here on github at the moment. Maybe once my 330 breaks I'll have to start fiddling around with one of the newer devices as well...

**Comment by ghost** [comment]
ok, here is the file that i received: [BRYTON.zip](https://github.com/andbue/ors2bryton/files/6028660/BRYTON.zip)

maybe it is useful and something can be done with it?

**Comment by matheus0312** [comment]
Hey guys,

I'm working on this problem also for my Rider420.

Have you guys had any updates in it?

I mostly already decoded the .smy, .tinfo, .zmap and .zinfo files. From what I discovered, the information in the .zinfo and .zmap files arenot rewritten everytime the route is loaded, and in my tests, and during the normal unit use, I noticed that sometimes bugs appear in the generated files which stops the turn by turn navigation before the end of the track. So having a way to generate the files manually may be a plus.

I'm currently working on the .fit file, which I just found out that is the most important route file latter. Indeed it does not follow the default .fit format specified by ant+ which is a shame. 

All my tests until now were done based on personalized .gpx files which I know the streets in which it passes and the number of turns. I plan to start "following the routes" created in order to verify what is displayed in the unit as proposed by @andbue.

Hope to have something new in the next weeks. In case you guys have news please contact me :)
